### ISSUE TITLE
Pagerduty not following new accounts

### ISSUE THREAD

**Issue by danielkasen** [elastic/elasticsearch#35601]
**Elasticsearch version** (`bin/elasticsearch --version`):
6.3.2

**Plugins installed**: []

**JVM version** (`java -version`):
OpenJDK 1.8.72

**OS version** (`uname -a` if on a Unix-like system):
Ubuntu 14.04

**Description of the problem including expected versus actual behavior**:
After adding a new pagerduty integration into the cluster settings, when you try to call it using the pagerduty watcher action it just sends the event to the default integration. 

**Steps to reproduce**:

Include a default pagerduty account in the elasticsearch.yml file:

```  
xpack:
  notification:
    pagerduty:
      account:
        monitoring:
          service_api_key: "XXXXXXX"
          event_defaults:
            description: "Watch Notification"
```

Then add more accounts via the _cluster/settings

```
PUT _cluster/settings
{
  "persistent": {
    "xpack": {
      "notification": {
        "pagerduty": {
          "default_account": "elasticsearch-logs",
          "account": {
            "monitoring": {
              "service_api_key": "asdsadasd"
            },
            "elasticsearch-logs": {
              "service_api_key": "asdasdadsa"
            },
            "kibana-logs": {
              "service_api_key": "asdasdasd"
            },
            "logstash-logs": {
              "service_api_key": "asdadsadas"
            }
          }
        }
      }
    }
  }
}
```
Then try to send an event to one of those accounts:

```
{
  "trigger": {
    "schedule": {
      "interval": "15m"
    }
  },
  "input": {
    "search": {
      "request": {
        "search_type": "query_then_fetch",
        "indices": [
          "elk-*"
        ],
        "types": [],
        "body": {
          "query": {
            "bool": {
              "filter": [
                {
                  "range": {
                    "@timestamp": {
                      "to": "{{ ctx.trigger.triggered_time }}",
                      "from": "{{ ctx.trigger.scheduled_time }}||-6m"
                    }
                  }
                },
                {
                  "match": {
                    "fileset.module": "curator"
                  }
                }
              ],
              "must": [
                {
                  "query_string": {
                    "query": "message: \"forceMerging index\"",
                    "default_field": "message"
                  }
                }
              ]
            }
          },
          "size": 100
        }
      }
    }
  },
  "condition": {
    "compare": {
      "ctx.payload.hits.total": {
        "lt": 1
      }
    }
  },
  **"actions": {
    "notify-pagerduty": {
        "throttle_period": "5m",
        "pagerduty": {
            "account": "elasticsearch-logs",
            "description": "STG - Force Merge Hasn't Run in 6 Hours"
        }
    }
  }**
}
```

THis results in the default `monitoring` account getting paged which is defined in the elasticsearch.yml file

**Comment by danielkasen** [comment]
So upon further inspection it would appear that ES 6.3 doesn't respect the cluster settings for PD at all. In order to send to a different endpoint you have to create the keystore record and restart every node. Then you can use the specific names.

**Comment by danielkasen** [comment]
This appears to be resolved in 6.5.4
